src/gltf-component-mappings.js

```diff
@@ -181,3 +189,75 @@ AFRAME.GLTFModelPlus.registerComponent("spawner", "spawner", (el, componentName,
   });
   el.setAttribute("hoverable", "");
 });
+
+function propertyMapper(componentKey, componentProperty, componentValue) {
+  const mappedComponent = AFRAME.GLTFModelPlus.components[componentKey];
+
+  if (!mappedComponent) {
+    return null;
+  }
+
+  const componentName = mappedComponent.componentName;
+
+  if (mappedComponent.propertyMapper) {
+    return mappedComponent.propertyMapper(componentName, componentProperty, componentValue);
+  }
+
+  const componentSchema = AFRAME.components[componentName].schema;
+
+  if (typeof componentValue === "object") {
+    if (!componentSchema[componentProperty] || !componentSchema[componentProperty].public) {
+      return null;
+    }
+  }
+
+  return { name: componentName, property: componentProperty, value: componentValue };
+}
+
+AFRAME.GLTFModelPlus.registerComponent(
+  "trigger-volume",
+  "trigger-volume",
+  (el, componentName, componentData, components, rootEl) => {
+    const {
+      size,
+      target,
+      enterComponent,
+      enterProperty,
+      enterValue,
+      leaveComponent,
+      leaveProperty,
+      leaveValue
+    } = componentData;
+
+    const enterAction = propertyMapper(enterComponent, enterProperty, enterValue);
+
+    if (!enterAction) {
```

Comment:
I disagree, these checks are for property sanitization. I'd like to keep them in one place so that if someone is editing the component they aren't confused and accidentally expose the wrong properties. See my point above about moving to a system that checks for a `public` attribute on the schema. I think that may be a better option for allowing you configure what properties are exposed in the component. I still think the system that checks that public properties are being set should live outside that component. Components should describe their public interfaces, but should not need to handle sanitization.

Reply:
This isn't about the public interface of the `trigger-volume` component, it is about that component caring about other components properties. Right now you could create a `trigger-volume` component directly and give it invalid properties since they are only checked by the inflator, so it makes sense to move that check upstream. As for the whole "public" property check, I wouldn't really bother with that personally, its not like we are doing an `eval` on the value or anything, so it doesnt scare me very much. I would rather find out novel ways people think to use things we didn't think of than manually have to whitelist things.